tests/e2e/nvidia-container-cli_test.go

```diff
@@ -0,0 +1,209 @@
+set -euo pipefail
+export DEBIAN_FRONTEND=noninteractive
+
+# Add Docker's official GPG key:
+apt-get update
+apt-get install -y ca-certificates curl apt-utils
+install -m 0755 -d /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" | \
+  tee /etc/apt/sources.list.d/docker.list > /dev/null
+apt-get update
+
+apt-get install -y docker-ce docker-ce-cli containerd.io
+
+# start dockerd in the background
+dockerd &
+
+# wait for dockerd to be ready
+while ! docker info > /dev/null 2>&1; do
+    echo "Waiting for dockerd to be ready..."
+    sleep 1
+done
+
+# install nvidia-container-toolkit from NVIDIA_TOOLKIT_IMAGE using docker
+TEMP_DIR="/tmp/nvidia-container-toolkit"
+mkdir -p "$TEMP_DIR"
+rm -f /usr/bin/nvidia-container-runtime-hook
+ln -s "$TEMP_DIR/toolkit/nvidia-container-runtime-hook" /usr/bin/nvidia-container-runtime-hook
+
+docker run --rm -d --name nvidia-container-toolkit-packaging --entrypoint "/usr/bin/sleep" ${NVIDIA_TOOLKIT_IMAGE}-packaging 120
+docker cp nvidia-container-toolkit-packaging:/artifacts/packages/ubuntu18.04/amd64 ${TEMP_DIR}
+docker rm -f nvidia-container-toolkit-packaging
+
+dpkg -i ${TEMP_DIR}/amd64/nvidia-container-toolkit-base_*_amd64.deb
+dpkg -i ${TEMP_DIR}/amd64/libnvidia-container1_*_amd64.deb
+dpkg -i ${TEMP_DIR}/amd64/libnvidia-container-tools_*_amd64.deb
+
+nvidia-container-cli --version
+
+apt-get update -y && apt-get install -y curl gnupg2
+
+# Create a temporary directory and rootfs path
+TMPDIR="$(mktemp -d)"
+ROOTFS="${TMPDIR}/rootfs"
+mkdir -p "${ROOTFS}"
+
+# Expose ROOTFS for the child namespace
+export ROOTFS
+
+curl http://cdimage.ubuntu.com/ubuntu-base/releases/22.04/release/ubuntu-base-22.04-base-amd64.tar.gz | tar -C $ROOTFS -xz
+
+# Enter a new mount + PID namespace so we can pivot_root without touching the
+# container's original filesystem.
+unshare --mount --pid --fork --propagation private -- sh -eux <<'IN_NS'
+  : "${ROOTFS:?}"
+
+  # 1 Bind-mount the new root and make the mount private
+  mount --bind "$ROOTFS" "$ROOTFS"
+  mount --make-private "$ROOTFS"
+  cd "$ROOTFS"
+
+  # 2 Minimal virtual filesystems
+  mount -t proc  proc proc
+  mount -t sysfs sys  sys
+  mount -t tmpfs tmp  tmp
+  mount -t tmpfs run  run
+
+  # 3 GPU setup via nvidia-container-cli
+  # Add potential install locations of nvidia-container-cli to PATH.
+  # /artifacts/{rpm,deb}/usr/bin are where the binary ends up after the
+  # packages are extracted in the application image.  /work is included for
+  # completeness since some images may copy the binary there.
+  export PATH="${PATH}:/artifacts/rpm/usr/bin:/artifacts/deb/usr/bin:/work"
```

Comment:
Is this still the case? We now install the `deb` packages directly, meaning these should be installed to `/usr/bin` which SHOULD be in the path already.

Reply:
Question: Why do we include the `/artifacts` paths here?

Reply:
Good catch, that's from the previous iteration when I was trying to run inside the toolkit image it self (Docker in docker) removed now